src/react.force.oauth.js

```diff
@@ -81,17 +81,19 @@ export const getAuthCredentials = (success, fail) => {
 /**
  * Logout the current authenticated user. This removes any current valid session token
  * as well as any OAuth refresh token.  The user is forced to login again.
- * This method does not call back with a success or failure callback, as 
+ * This method does not call back with a success or failure callback, as
  * (1) this method must not fail and (2) in the success case, the current user
  * will be logged out and asked to re-authenticate.  Note also that this method can only
  * be called once per page load.  Initiating logout will ultimately redirect away from
  * the given page (effectively resetting the logout flag), and calling this method again
  * while it's currently processing will result in app state issues.
  */
 export const logout = () => {
+    // only set callback for android, since iOS will not invoke callback.
+    const logoutCb = SalesforceOauthReactBridge ? () => { logoutInitiated = false; } : null;
     if (!logoutInitiated) {
         logoutInitiated = true;
-        exec(null, null, "logoutCurrentUser", {});
+        exec(logoutCb, null, "logoutCurrentUser", {});
     }
```

Comment:
allow callback in android land to be able to reset the logoutInitiated flag, to allow for more that 1 logout without reloading the app. See https://github.com/forcedotcom/SalesforceMobileSDK-Android/pull/1679/files for android changes

Reply:
@gla5001 What happens on `iOS` when you attempt multiple logouts? Sounds like we should be resetting `logoutInitiated` on `iOS` as well?

Reply:
It sounds like we should be resetting the context for every logout attempt on `Android` as well (not just the first logout), which will automatically reset `logoutIntitiated`, as opposed to providing a callback. The fix should probably be in our native `Android` library.

Reply:
@bhariharan the iOS behaves as it should with multiple logouts, the react context is correctly reset along with logoutInitiated. This is specifically an android issue. I spent some time debugging and looking into the native android code, specifically the sdkManager, and could not find a solution that resets the context on logout. The only thing that worked for me was providing a callback and calling the callback in native code. I could be overlooking an area though, any idea where this would be happening?

Reply:
@gla5001 Check out my comment [here](https://github.com/forcedotcom/SalesforceMobileSDK-ReactNative/issues/72#issuecomment-383318948). Could you take a look at our sample app - maybe that'll shed some light?

Reply:
Sure. I’ll check it out and get back to you early in the week